bin/rebuild-package.sh

```diff
@@ -0,0 +1,320 @@
+#!/bin/bash
+
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+###
+### This script represents a gitlab-ci job, corresponding to a single release
+### spec.  As such this script must first decide whether or not the spec it
+### has been assigned is up to date on the remote binary mirror.  If it is
+### not (i.e. the source code has changed in a way that caused a change in the
```

Comment:
Continuing https://github.com/spack/spack/pull/10274#discussion_r253725186

If the source code changes by one character, that will change the hash of the archive (we specifically use hash algorithms that strive for this property), which is part of the full hash. Therefore I think it is clearer to say:

> if the source code has changed

and exclude "in a way that...". Since all changes should produce a different hash.